---
version: "3.4"
services:
  vault:
    image: hashicorp/vault:latest@sha256:1262354cd28697b7982ea3b9b6f159a996bdaad0b5270765e31b67797ce15bea
    volumes:
      - "{{ docker_volume_dirs.vault[0] }}:/vault/config"
      - "{{ docker_volume_dirs.vault[1] }}:/vault/file"
      - "{{ docker_volume_dirs.vault[2] }}:/vault/logs"
      - "{{ docker_volume_dirs.vault[3] }}:/vault/plugins"
    ports:
      - "{{ vault.port }}:{{ vault.internal_port }}"
    cap_add:
      - IPC_LOCK
    restart: always
    command: [vault, server, -config, /vault/config/config.hcl]
  #   healthcheck:
  #     test: [CMD, wget, "{{ healthchecks.healthcheck }}", -O, /dev/null]
  #     timeout: 45s
  #     interval: "{{ healthchecks_defaults.ping.interval }}s"
  #     retries: 5
  opconnect_api:
    image: 1password/connect-api:latest@sha256:1695b2fdc6897e49fdfd80e64bdb78f4c64f7691a8d46957c0372e47f776430e
    volumes:
      - "{{ docker_volume_dirs.opconnect[0] }}/1password-credentials.json:/home/opuser/.op/1password-credentials.json"
      - data:/home/opuser/.op/data
    ports:
      - "{{ opconnect.api.port }}:{{ opconnect.api.internal_port }}"
    restart: always
  opconnect_sync:
    image: 1password/connect-sync:latest@sha256:d5e937b2b7e314dbbaffacf469dc8f30376b5954a3ad836c8d96e6cb579d7fab
    volumes:
      - "{{ docker_volume_dirs.opconnect[0] }}/1password-credentials.json:/home/opuser/.op/1password-credentials.json"
      - data:/home/opuser/.op/data
    ports:
      - "{{ opconnect.sync.port }}:{{ opconnect.sync.internal_port }}"
    restart: always

volumes:
  data:
